sherpa/astro/data.py

```diff
@@ -250,6 +251,9 @@ def _validate_energy_ranges(self, label, elo, ehi, ethresh):
 
         return elo, ehi
 
+    def _get_data_space(self, filter=False):
```

Comment:
Indeed. I'll do it.

Incidentally, it would be nice to come up with good documentation on the existing Data classes and methods, even if it means to reverse-engineer them to understand which methods are used when. And that might call for some deep refactoring. There is some "clever" code that makes things rather unreadable and the hierarchy complex to browse and reuse.

Reply:
One of the issues that helped delay the sphinx docs was a lot of undocumented code!

Reply:
could you please double check the formatting is OK?

Reply:
LGTM (as it's an internal style routine I am not too bothered with the formatting just yet ;-)